src/extractors/withTranslationHOC.ts

```diff
@@ -251,7 +265,7 @@ function findTFunctionCallsInClassComponent(
         // We have something in the form "this.props.t". Let's see if it's an
         // actual function call or an assignment.
         const tExpression = path.parentPath.parentPath.parentPath;
-        if (tExpression.isCallExpression()) {
+        if (isCallee(path.parentPath.parentPath, tExpression)) {
```

Comment:
This feels even weirder now :confused: . I think it would be clearer if `tExpression` was not the expected `callExpression` but the actual complete t identifier like so:

```typescript
        // We have something in the form "this.props.t". Let's see if it's an
        // actual function call or an assignment.
        const tExpression = path.parentPath.parentPath;
        if (isCallee(tExpression)) {
          // Simple case. Direct call to "this.props.t()"
          result.push(tExpression.parentPath);
        } else if (tExpression.parentPath.isVariableDeclarator()) {
          // Hard case. const t = this.props.t;
          // Let's loop through all references to t.
          const id = tExpression.parentPath.get('id');
          if (!id.isIdentifier()) return;
          for (const reference of id.scope.bindings[id.node.name]
            .referencePaths) {
            if (isCallee(reference)) {
              result.push(reference.parentPath);
            }
          }
        }
```

Reply:
Done.